presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -360,38 +367,88 @@ protected Object visitSearchedCaseExpression(SearchedCaseExpression node, Object
             return result;
         }
 
+        private Object processWithExceptionHandling(Expression expression, Object context)
+        {
+            if (expression == null) {
+                return null;
+            }
+
+            try {
+                return process(expression, context);
+            }
+            catch (RuntimeException e) {
+                // HACK
+                // Certain operations like 0 / 0 or likeExpression may throw exceptions.
+                // Wrap them a FunctionCall and add it to the expression evaluation
+                return createFailureFunction(type(expression), e);
+            }
+        }
+
+        /**
+         * @return updated WhenClause or evaluated object. Optional.empty() if we can ignore this WhenClause
```

Comment:
`@return updated WhenClause, the result value if clause is always true, or Optional.empty() the clause is always false,`